p2p/peer.py

```diff
@@ -283,6 +283,10 @@ def close(self):
         self.reader.feed_eof()
         self.writer.close()
 
+    @property
+    def connected(self):
+        return not self.reader.at_eof()
```

Comment:
This worries me as a peer may be connected but no longer running. This may not be necessary with the changes I suggest above, but if something like it really is needed, we should consider using peer.is_running instead.